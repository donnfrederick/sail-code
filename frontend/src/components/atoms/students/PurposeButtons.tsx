import * as SessionsModels from 'models/sessions'
import * as StudentsModels from 'models/students'
import * as R from 'ramda'
import * as React from 'react'
import styled from 'styled-components'

interface Props {
  info: StudentsModels.Info
  purposes: SessionsModels.Purpose[]
  register(info: StudentsModels.Info): void
}

export default (props: Props) => {
  const { purposes, info, register } = props

  const registInterest = (event: any) => {
    const value = event.target.textContent
    const targetPurpose = R.find(R.propEq('name', value))(purposes)
    info.purposes = R.contains(targetPurpose, info.purposes)
      ? R.filter(purpose => !R.equals(targetPurpose, purpose), info.purposes)
      : info.purposes.concat(targetPurpose)

    register(info)
  }

  return (
    <Buttons>
      {purposes.map(purpose => (
        <PurposeButton
          key={purpose.id}
          data-checked={R.contains(purpose, info.purposes)}
          onClick={registInterest}
        >
          {purpose.name}
        </PurposeButton>
      ))}
    </Buttons>
  )
}

const Buttons = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-bottom: 100px;
`

const PurposeButton = styled.button`
  appearance: none;
  width: 100%;
  height: 88px;
  box-sizing: border-box;
  margin: 0 0 32px;
  padding: 0 40px;
  outline: none;
  border: 2px solid #138efd;
  border-radius: 44px;
  background-color: #f6f7fb;
  background-image: none;
  font-size: 28px;
  font-weight: 500;
  line-height: 84px;
  text-align: center;
  color: #138efd;

  &[data-checked='true'] {
    background-image: linear-gradient(293deg, #2eb1ff, #138efd);
    color: #ffffff;
  }
`
