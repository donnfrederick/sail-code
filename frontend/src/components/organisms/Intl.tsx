import React, {
  PropsWithChildren,
  Dispatch,
  SetStateAction,
  useContext,
  useState
} from 'react'
import { IntlProvider } from 'react-intl'
import { messages, browserLocale, Locale } from 'utils/intl'

export type LocaleState = {
  locale: Locale
}

const initialState: LocaleState = {
  locale: browserLocale
}

const IntlStateContext = React.createContext<LocaleState>(initialState)

const SetLocaleStateContext = React.createContext<
  Dispatch<SetStateAction<LocaleState>>
>(() => {
  // empty block
})

export function useLocaleState() {
  return useContext(IntlStateContext)
}
export function useSetLocaleState() {
  return useContext(SetLocaleStateContext)
}

export function LocaleStateProvider(props: {
  initialState?: LocaleState
  children: React.ReactNode
}) {
  const [state, setState] = useState<LocaleState>(
    props.initialState || initialState
  )
  return (
    <IntlStateContext.Provider value={state}>
      <SetLocaleStateContext.Provider value={setState}>
        {props.children}
      </SetLocaleStateContext.Provider>
    </IntlStateContext.Provider>
  )
}

export const Intl = (props: PropsWithChildren<{}>) => {
  const localeState = useLocaleState()

  return (
    <LocaleStateProvider>
      <IntlProvider
        messages={messages[localeState.locale]}
        locale={localeState.locale}
        defaultLocale="en"
      >
        {props.children}
      </IntlProvider>
    </LocaleStateProvider>
  )
}
