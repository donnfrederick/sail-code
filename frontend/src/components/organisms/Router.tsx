import * as Sentry from '@sentry/browser'
import ErrorPage from 'components/pages/Error'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider, Store } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { compose, lifecycle, withState } from 'recompose'
import createStore from 'store/createStore'
import resetViewport from 'utils/resetViewport'
import userAgent from 'utils/userAgent'

declare const gtag: any

export const history = createBrowserHistory()
export const store = createStore(history) as Store<any>

history.listen(() => {
  window.scrollTo(0, 0)
  gtag('config', 'UA-105776977-2', { page_path: history.location.pathname })
  userAgent()
  resetViewport()
})

interface Props {
  children: JSX.Element
  errors: Error | null
  setErrors: (state: any) => any
}

const isProd = process.env.NODE_ENV === 'production'

const sentry = compose(
  withState('errors', 'setErrors', null),
  isProd
    ? lifecycle<Props, {}, {}>({
        componentDidCatch(error, errorInfo) {
          this.props.setErrors(error)
          Sentry.withScope(scope => {
            Object.keys(errorInfo).forEach(key => {
              scope.setExtra(key, errorInfo[key])
            })
            Sentry.captureException(error)
          })
        }
      })
    : lifecycle<Props, {}, {}>({
        componentDidCatch(error) {
          this.props.setErrors(error)
        }
      })
)

export default sentry((props: Props) => {
  const { children, errors } = props

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {errors ? <ErrorPage /> : children}
      </ConnectedRouter>
    </Provider>
  )
})
