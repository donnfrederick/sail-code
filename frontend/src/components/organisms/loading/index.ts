import Loading from 'components/organisms/loading/Loading'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  isFetchingConversation: boolean
  isFetchingLocations: boolean
  isFetchingNotification: boolean
  isFetchingOrganizations: boolean
  isFetchingStudentsMe: boolean
  isFetchingTeachersMe: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    isFetchingConversation: state.rootReducer.conversation.fetching > 0,
    isFetchingLocations: state.rootReducer.locations.isFetching,
    isFetchingNotification: state.rootReducer.notification.isFetching,
    isFetchingOrganizations: state.rootReducer.organizations.isFetching,
    isFetchingStudentsMe: state.rootReducer.students.isFetching,
    isFetchingTeachersMe: state.rootReducer.teachers.isFetching
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {}
)

export default enhancer(Loading)
