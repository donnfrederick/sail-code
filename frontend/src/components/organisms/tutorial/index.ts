import Tutorial from 'components/organisms/tutorial/Tutorial'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'
import * as TutorialActions from 'reducers/tutorial'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  shouldShow: boolean
  showConfirmation: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    shouldShow: state.rootReducer.tutorial.shouldShow,
    showConfirmation: state.rootReducer.teachers.showConfirmation
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {
    hide: TutorialActions.hide,
    show: TutorialActions.show
  }
)

export default enhancer(Tutorial)
