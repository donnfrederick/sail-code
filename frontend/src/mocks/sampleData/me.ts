export const me = {
  absence: 0,
  auth_token: '',
  birthday: '',
  conversation_level: 0,
  conversations: [],
  country: '',
  desired_condition: 0,
  email: '',
  evaluate: {
    1: 0,
    2: 0,
    3: 0,
    4: 0
  },
  grade: 'member',
  highly_reliable: false,
  hobbies: [],
  id: 0,
  lateness: 0,
  level: 0,
  location: '',
  name: '',
  name_ja: '',
  organization_device: null,
  organization_sections: [],
  organizations: [],
  payment_state: 'hidden',
  picture_url: '',
  purposes: [],
  rated_conversation_level: 1,
  sex: 0,
  timezone: '',
  type: '',
  username: '',
  web_socket_token: '',
  introduce: ''
}

export const info = {
  email: '',
  password: '',
  current_password: '',
  conversation_level: 0,
  level: 0,
  country: '',
  timezone: '',
  name: '',
  name_ja: '',
  gender: 0,
  picture: '',
  grade: 'member',
  hobbies: [],
  purposes: [],
  categories: [],
  tags: [],
  selected_tags: [],
  added_tags: [],
  desired_condition: 0,
  introduce: '',
  sysp_nl: ''
}
