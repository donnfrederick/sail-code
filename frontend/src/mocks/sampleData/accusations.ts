import { conversation as sampleConversation } from 'mocks/sampleData/conversation'
import { me } from 'mocks/sampleData/me'
import * as AccusationsModels from 'models/accusations'

export const reasons: AccusationsModels.Reason[] = [
  {
    id: 1,
    name: 'reason 1'
  },
  {
    id: 2,
    name: 'reason 2'
  },
  {
    id: 3,
    name: 'reason 3'
  }
]

export const accusationsResponse: AccusationsModels.AccusationsResponse = {
  conversation: sampleConversation,
  from_user: me,
  to_user: me
}
