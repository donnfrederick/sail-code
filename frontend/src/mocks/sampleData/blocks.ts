import { userProfile as user } from 'mocks/sampleData/userProfile'
import * as BlocksModels from 'models/blocks'

export const blocks: BlocksModels.GetBlocksResponse = {
  meta: {
    current_page: 0,
    per_page: 2,
    previous_page: 1,
    next_page: 3,
    total_pages: 5,
    total_entries: 10
  },
  data: [user]
}

export const blockedUser = user
