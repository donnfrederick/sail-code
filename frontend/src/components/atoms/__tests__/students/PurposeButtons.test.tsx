import PurposeButtons from 'components/atoms/students/PurposeButtons'
import * as sampleData from 'mocks/sampleData/me'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Intl } from 'components/organisms/Intl'

test('render <PurposeButtons />', () => {
  const tree = renderer
    .create(
      <Intl>
        <PurposeButtons
          purposes={[]}
          info={sampleData.info}
          register={() => null}
        />
      </Intl>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
