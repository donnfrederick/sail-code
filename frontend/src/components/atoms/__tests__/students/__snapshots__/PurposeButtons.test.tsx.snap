// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <PurposeButtons /> 1`] = `
<div
  className="sc-bdVaJa gHvLkt"
/>
`;
